openchain/consensus/obcpbft/obc-sieve.go

```diff
@@ -36,29 +36,52 @@ type obcSieve struct {
 	stack consensus.Stack
 	pbft  *pbftCore
 
+	startup chan []byte
+
 	id            uint64
 	epoch         uint64
 	imminentEpoch uint64
 	blockNumber   uint64
 	currentReq    string
-	currentResult []byte
+	currentView   uint64
+	lowWaterSeqNo uint64
 
-	currentTx   []*pb.Transaction
 	verifyStore []*Verify
 
 	queuedExec map[uint64]*Execute
 	queuedTx   [][]byte
+
+	executor        Executor
+	validResultChan chan *validResult
+}
+
+type validResult struct {
+	commit  bool
+	sieveID []byte
+	peerIDs []*pb.PeerID
 }
 
 func newObcSieve(id uint64, config *viper.Viper, stack consensus.Stack) *obcSieve {
 	op := &obcSieve{stack: stack, id: id}
 	op.queuedExec = make(map[uint64]*Execute)
-	op.pbft = newPbftCore(id, config, op, stack)
-	op.pbft.sts.RegisterListener(op)
+	op.validResultChan = make(chan *validResult)
+	op.startup = make(chan []byte)
+
+	op.executor = NewOBCExecutor(config, op, stack)
+
+	logger.Debug("Replica %d obtaining startup information", id)
+	startupInfo := <-op.startup
+	close(op.startup)
+
+	op.pbft = newPbftCore(id, config, op, startupInfo)
 
 	return op
 }
 
+func (op *obcSieve) Startup(seqNo uint64, id []byte) {
```

Comment:
This is a little hacky, and I don't like the code duplication.  I considered having this value returned via the `NewOBCExecutor`, but chose not to for a few different reasons.

In design, I tried to think of `Exector` and the `Orderer` as being two different processes, talking via the network.  Because of this, it makes sense that the instantiation of the `OBCExecutor` should not require a connection to the `Orderer`, similarly, the fact that the `Orderer` is invoking the constructor of the `Executor` is a local implementation detail, so relying on output from its instantiation seemed wrong.  You can think of that `Startup` method more as a network `Handshake` method, logically.

If there are strong feelings that this is the wrong approach, I can rework the patch to provide the seqNo and ID via the constructor call.